cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs

```diff
@@ -633,7 +633,11 @@ runQueryPoolStateCmd
 
         beo <- requireEon BabbageEra era
 
-        result <- lift (queryPoolState beo $ Just $ Set.fromList poolIds)
+        let poolFilter = case allOrOnlyPoolIds of
+              All -> Nothing
+              Only poolIds -> Just $ Set.fromList poolIds
+
+        result <- lift (queryPoolState beo poolFilter)
```

Comment:
Where did you see that passing `Nothing` means "no-filtering"? Looking up the definition of `QueryPoolState`, it's not documented. (I don't see what other behavior `Nothing` could have, but being curious).